### PR TITLE
 #780-[ENHANCEMENT] Replace "findViewById" by "@BindView"

### LINKED ISSUE #780: [ENHANCEMENT] Replace "findViewById" by "@BindView"
**Is your feature request related to a problem? Please describe.**
I noticed that you're using Butterknife but there is still some "findViewById" usage here and there, sometimes on classes that already use ButterKnife.

Example from the "CurrencyListViewActivity" class : 
```
@BindView(R.id.listView)
RecyclerView mListview;

[...]

mListview = findViewById(R.id.listView);
```

**Describe the solution you'd like**
Take some time to tidy up those classes.

### PR DESCRIPTION
**Pull request by hafiz703**
## Description

Fixed most implementations of findviewById to @BindView in activities except those in ViewGroups / Fragments.

Fixes #780 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x ] `./gradlew assembleDebug assembleRelease`
- [x ] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 1]
```diff
@@ -175,17 +183,17 @@ protected void onCreate(Bundle savedInstanceState) {
...
         MenuItem menuItem = menu.findItem(resId);
         menuItem.setChecked(true);
     }
 
     @Override
     public void onBackPressed() {
-        DrawerLayout drawer = findViewById(R.id.drawer_layout);
-        if (drawer != null && drawer.isDrawerOpen(GravityCompat.START)) {
-            drawer.closeDrawer(GravityCompat.START);
+        //DrawerLayout drawer = findViewById(R.id.drawer_layout);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: dont comment. Remove it.

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 2]
```diff
@@ -39,7 +39,9 @@
 import java.io.IOException;
 import java.util.Objects;
 
+import butterknife.BindView;
 import butterknife.ButterKnife;
+import butterknife.Optional;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: is this import needed?

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 3]
```diff
@@ -87,21 +89,27 @@
...
     private Handler mHandler;
-    private NavigationView mNavigationView;
     private int mPreviousMenuItemId;
     private static final String travelShortcut = "io.github.project_travel_mate.TravelShortcut";
     private static final String myTripsShortcut = "io.github.project_travel_mate.MyTripsShortcut";
     private static final String utilitiesShortcut = "io.github.project_travel_mate.UtilitiesShortcut";
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.drawer_layout)
+    DrawerLayout mDrawer;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: if variables are not private, no need to follow this naming convention.
Remove `m` from start

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: `drawerLayout`

File: Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 import com.eftimoff.androipathview.PathView;
 
+import butterknife.BindView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java]: is it needed? It isn't being used anywhere here

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 5]
```diff
@@ -72,8 +73,11 @@
     LottieAnimationView animationView;
     @BindView(R.id.cities_list)
     ListView lv;
+    @BindView(R.id.search_view)
+    MaterialSearchView mMaterialSearchView;
 
-    private MaterialSearchView mMaterialSearchView;
+
+//    private MaterialSearchView mMaterialSearchView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: dont comment. remove

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 6]
```diff
@@ -72,8 +73,11 @@
     LottieAnimationView animationView;
     @BindView(R.id.cities_list)
     ListView lv;
+    @BindView(R.id.search_view)
+    MaterialSearchView mMaterialSearchView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: change var name. Do it for other places too

File: Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java [hunk 7]
```diff
@@ -51,6 +51,7 @@
     @BindView(R.id.et_query)
     EditText etQuery;
 
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java]: remove this file from commit

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java [hunk 8]
```diff
@@ -81,7 +84,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_hotels);
 
-        Toolbar toolbar = findViewById(R.id.toolbar);
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: remove this extra line

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java [hunk 9]
```diff
@@ -40,8 +40,7 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_conversion_listview);
 
         currences_names = new ArrayList<>();
-        mListview = findViewById(R.id.listView);
-        mCurrencySearch = findViewById(R.id.currencySearch);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java]: you havent initialized this using butterknife?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java [hunk 10]
```diff
@@ -119,7 +119,7 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     public void createCode() {
 
-        ImageView qrCodeView = findViewById(R.id.im);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: where is this being initialized?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java [hunk 11]
```diff
@@ -102,4 +102,4 @@ private void onItemAutocompleteItemSelected() {
         this.setSelectedText(TimeZone.getTimeZone(this.mAutoCompleteTextViewTimezone.getText().toString()));
     }
 
-}
+}
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: remove this file

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/TimezoneListViewActivity.java [hunk 12]
```diff
@@ -39,8 +39,7 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_timezone_listview);
 
         timezone_names = new ArrayList<>();
-        mListview = findViewById(R.id.listView);
-        mTimezoneSearch = findViewById(R.id.timezoneSearch);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/TimezoneListViewActivity.java]: where is this being initialized?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java (hunk 9 above)
Comment (hafiz703) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java]: Yes, above onCreate

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 13]
```diff
@@ -87,21 +89,27 @@
...
     private Handler mHandler;
-    private NavigationView mNavigationView;
     private int mPreviousMenuItemId;
     private static final String travelShortcut = "io.github.project_travel_mate.TravelShortcut";
     private static final String myTripsShortcut = "io.github.project_travel_mate.MyTripsShortcut";
     private static final String utilitiesShortcut = "io.github.project_travel_mate.UtilitiesShortcut";
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.drawer_layout)
+    DrawerLayout Drawer;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: variable names shouldnt start with capital letter
`drawerlayout`

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 14]
```diff
@@ -87,21 +89,27 @@
...
+    @BindView(R.id.nav_view)
+    NavigationView NavigationView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: here also

File: Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java [hunk 15]
```diff
@@ -19,6 +17,7 @@
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java]: I dont see any relevant change in this file. Please remove this from commit

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 16]
```diff
@@ -72,8 +73,11 @@
     LottieAnimationView animationView;
     @BindView(R.id.cities_list)
     ListView lv;
+    @BindView(R.id.search_view)
+    MaterialSearchView MaterialSearchView;
 
-    private MaterialSearchView mMaterialSearchView;
+
+//    private MaterialSearchView MaterialSearchView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: remove this line

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 17]
```diff
@@ -72,8 +73,11 @@
     LottieAnimationView animationView;
     @BindView(R.id.cities_list)
     ListView lv;
+    @BindView(R.id.search_view)
+    MaterialSearchView MaterialSearchView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: change for all places - shouldnt start with capital letter

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 18]
```diff
@@ -89,7 +89,8 @@
     BottomSheetBehavior sheetBehavior;
     @BindView(R.id.bottom_sheet)
     LinearLayout layoutBottomSheet;
-    private MapView mMap;
+    @BindView(R.id.map)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: this change was intended for all places. Only Private variables start with `m...`

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java [hunk 19]
```diff
@@ -71,6 +72,8 @@
     LinearLayout mLinearLayout;
     @BindView(R.id.animation_view)
     LottieAnimationView animationView;
+    @BindView(R.id.activityAddNewTrip)
+    CoordinatorLayout activityAddNewTrip;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java]: `layoutAddNewTrip`

File: Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java (hunk 7 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java]: Not resolved.

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java (hunk 8 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: dont mark the conversation resolved, if you havent taken any action against it

File: Android/app/src/main/java/io/github/project_travel_mate/travel/ListViewRealTimeActivity.java [hunk 20]
```diff
@@ -30,6 +30,8 @@
     TabLayout tabLayout;
     @BindView(R.id.viewpager)
     ViewPager viewPager;
+    @BindView(R.id.action_map_view)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/ListViewRealTimeActivity.java]: change name

File: Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java [hunk 21]
```diff
@@ -69,12 +69,25 @@
...
     @BindView(R.id.layout)
     LinearLayout layout;
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.item_title)
+    TextView title;
+    @BindView(R.id.item_description)
+    TextView description;
+    @BindView(R.id.map)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java]: change name

File: Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java [hunk 22]
```diff
@@ -85,13 +98,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        Toolbar toolbar = findViewById(R.id.toolbar);
+
         setSupportActionBar(toolbar);
 
         ButterKnife.bind(this);
 
         mHandler = new Handler(Looper.getMainLooper());
-        mMap = findViewById(R.id.map);
+//        mMap = findViewById(R.id.map);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java]: remove this line

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java (hunk 10 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: ??

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java (hunk 11 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: ??

File: Android/app/src/main/java/utils/TravelMateAnalogClock.java [hunk 23]
```diff
@@ -274,58 +280,55 @@ private void main(Context ctx) {
 
         inflate(ctx, R.layout.analog_clock, this);
 
-        mAnalogFace = findViewById(R.id.face);
-        mAnalogHour = findViewById(R.id.hour);
-        mAnalogMinute = findViewById(R.id.minute);
-        mAnalogSecond = findViewById(R.id.second);
+//
```
Comment (Swati4star) [inline Android/app/src/main/java/utils/TravelMateAnalogClock.java]: remove line

File: Android/app/src/main/java/utils/TravelMateAnalogClock.java [hunk 24]
```diff
@@ -47,6 +43,16 @@
...
     private Context mCtx;
 
+    @BindView(R.id.face)
+    AppCompatImageView AnalogFace;
+    @BindView(R.id.hour)
+    AppCompatImageView AnalogHour;
+    @BindView(R.id.minute)
+    AppCompatImageView AnalogMinute;
+    @BindView(R.id.second)
+    AppCompatImageView AnalogSecond;
```
Comment (Swati4star) [inline Android/app/src/main/java/utils/TravelMateAnalogClock.java]: no variable name should start with capital letter

**Comment by prabhakar267** [comment]
Closing this due to inactivity. @hafiz703 reopen this if you address the comments.

**Pull request closed without merging**